examples/upscale_cond.cpp

```diff
@@ -74,109 +65,129 @@
+            << "                                     0 if there are no mud types.\n"
+            << "                                     Default 0\n"
+            << "  -mud2rocktype <int>             -- Second mud rock type. Default 0\n"
+            << "  -jFunctionCurve <int>           -- Which column in the supplied J-files\n"
+            << "                                     corresponds to the J-function values.\n"
+            << "                                     Default 2.\n"
+            << "  -surfaceTension <float>         -- Surface tension to use in J-function/Pc conversion.\n"
+            << "                                     Default 11 dynes/cm (oil-water systems). In absence of\n"
+            << "                                     a correct value, the surface tension for gas-oil systems\n"
+            << "                                     could be 22.5 dynes/cm.\n"
+            << "  -interpolate <integer>          -- If supplied and > 1, the output data points will be\n"
+            << "                                     interpolated using monotone cubic interpolation\n"
+            << "                                     on a uniform grid with the specified number of\n"
+            << "                                     points. Suggested value: 1000.\n\n"
+            << "  -rock<int>cemexp <float>        -- Cementation exponent can be set on a per rocktype basis\n"
+            << "  -rock<int>satexp <float>        -- Saturation exponent can be set on a per rocktype basis\n\n"
+            << "Jfunctions are data files with two colums of numbers. The first column is water\n"
+            << "saturation, the second is the J-function. The first Jfunction, Jfunc1.data\n"
+            << "corresponds to the first rock type defined in the eclipsefile's SATNUM. The\n"
+            << "second correspond to the second rock type and so on. If just one Jfunc is\n"
+            << "given, this is used for all rock types" << std::endl;
+
+            // "minPoro" intentionally left undocumented
 
-}
+    }
 
-void usageandexit() {
-    usage();
-    exit(1);
-}
-// Assumes that permtensor_t use C ordering.
-double getVoigtValue(const SinglePhaseUpscaler::permtensor_t& K, int voigt_idx)
-{
-    assert(K.numRows() == 3 && K.numCols() == 3);
-    switch (voigt_idx) {
-    case 0: return K.data()[0];
-    case 1: return K.data()[4];
-    case 2: return K.data()[8];
-    case 3: return K.data()[5];
-    case 4: return K.data()[2];
-    case 5: return K.data()[1];
-    case 6: return K.data()[7];
-    case 7: return K.data()[6];
-    case 8: return K.data()[3];
-    default:
-        std::cout << "Voigt index out of bounds (only 0-8 allowed)" << std::endl;
-        throw std::exception();
+    void usageandexit() {
+        usage();
+        std::exit(EXIT_FAILURE);
     }
-}
 
+    // Assumes that permtensor_t use C ordering.
+    double getVoigtValue(const SinglePhaseUpscaler::permtensor_t& K, int voigt_idx)
```

Comment:
not a requirement but while at it this could use the methods in RelPermUtils i believe. these functions are/were copy-pasted all over the place.